flamby/datasets/fed_camelyon16/dataset_creation_scripts/tiling_slides.py

```diff
@@ -20,12 +22,44 @@
 from flamby.utils import read_config, write_value_in_config
 
 
+def add_borders(slide: Slide, width: int = 10):
+    """Add borders to a slide thumbnail of the mean slide color.
+
+    This helps removing artefacts on the borders and prevents histolab TissuMask to
+    make incorrect detections.
+    """
+    pixels = slide.thumbnail.load()
+    pixel_values = np.zeros((slide.thumbnail.size[0], slide.thumbnail.size[1], 3))
+    for i in range(slide.thumbnail.size[0]):
+        for j in range(slide.thumbnail.size[1]):
+            pixel_values[i, j, :] = list(pixels[i, j])
+    mean_value = tuple(int(np.mean(pixel_values[:, :, i])) for i in range(3))
```

Comment:
You can probably vectorize this list comprehension in numpy doing `np.mean(pixel_values, axis=2)` or something along this line.

Reply:
sorry there was a typo in my last comment !